aya/src/bpf.rs

```diff
@@ -534,6 +534,24 @@ impl Bpf {
     pub fn programs(&self) -> impl Iterator<Item = &Program> {
         self.programs.values()
     }
+
+    /// An iterator mutably referencing all the programs.
+    ///
+    /// # Examples
+    /// ```no_run
+    /// # let mut bpf = aya::Bpf::load(&[])?;
+    /// for program in bpf.programs_mut() {
+    ///     println!(
```

Comment:
I think the example should mutate the programs, so that it makes sense to iterate over mutable refs